app/controllers/projects_controller.rb

```diff
@@ -46,15 +46,30 @@ def new
   def create
     @project = Project.new(project_params)
     @project.current_user = current_user
+    saved = @project.save
 
-    if @project.save
+    if saved
       if Rails.application.config.samson.project_created_email
         ProjectMailer.created_email(@current_user, @project).deliver_now
       end
-      redirect_to @project
       Rails.logger.info("#{@current_user.name_and_email} created a new project #{@project.to_param}")
-    else
-      render :new
+    end
+
+    respond_to do |format|
+      format.html do
+        if saved
+          redirect_to @project
+        else
+          render :new
+        end
+      end
+      format.json do
+        if saved
+          render_as_json :project, @project
+        else
+          render_as_json :errors, @project.errors, status: :bad_request
```

Comment:
should be a 422

Reply:
actually these should be `render_json_error 422, exception.record.errors` for consistency
also this is getting kinda repetitive, so some kind of method extraction would be great :)